lib/decompress/zstd_decompress.c

```diff
@@ -2620,6 +2633,7 @@ size_t ZSTD_initDStream_usingDict(ZSTD_DStream* zds, const void* dict, size_t di
 {
     DEBUGLOG(4, "ZSTD_initDStream_usingDict");
     zds->streamStage = zdss_init;
+    zds->noForwardProgress = 0;
```

Comment:
Does `ZSTD_initDCtx_internal()` not get called during initialization?

Reply:
I don't think so. `initDCtx` is only invoked when creating a new context.

Re-using a context triggers a much lighter context init.
For `initDStream`, it's entirely contained in this rather short function.